jose/jws.py

```diff
@@ -161,7 +159,8 @@ def _sign_header_and_claims(encoded_header, encoded_claims, algorithm, key_data)
     signing_input = b'.'.join([encoded_header, encoded_claims])
     try:
         key = jwk.construct(key_data, algorithm)
-        signature = key.sign(signing_input)
+        dat = signing_input.decode('utf8').encode('utf8')
```

Comment:
Nit: This could use a comment explaining why it's being round-tripped through the decoder/encoder